File: apps/nihongo-android/app/src/main/java/com/edu/nihongo/domain/entity/Entities.kt
package com.edu.nihongo.domain.entity

enum class SyncStatus {
    SYNCED,
    PENDING,
    CONFLICT,
}

data class Vocabulary(
    val id: Long,
    val lessonNumber: Int,
    val kana: String,
    val kanji: String?,
    val meaning: String,
    val romaji: String,
    val sortOrder: Int,
    val syncStatus: SyncStatus = SyncStatus.SYNCED,
)

data class SrsCard(
    val id: Long,
    val vocabularyId: Long,
    val easeFactor: Float,
    val interval: Int,
    val repetitions: Int,
    val nextReviewAt: Long,
    val mastered: Boolean,
    val syncStatus: SyncStatus = SyncStatus.SYNCED,
    val updatedAt: Long = System.currentTimeMillis(),
)

data class ReviewCard(
    val card: SrsCard,
    val kana: String,
    val kanji: String?,
    val meaning: String,
    val romaji: String,
)

data class OverlayLabel(
    val rect: android.graphics.RectF,
    val original: String,
    val translated: String,
)

File: apps/nihongo-mobile/src/domain/entities.ts
export type SyncStatus = 'synced' | 'pending' | 'conflict';

export interface Vocabulary {
  id: number;
  lessonNumber: number;
  kana: string;
  kanji: string | null;
  meaning: string;
  romaji: string;
  sortOrder: number;
  syncStatus: SyncStatus;
}

export interface SrsCard {
  id: number;
  vocabularyId: number;
  easeFactor: number;
  interval: number;
  repetitions: number;
  nextReviewAt: number;
  mastered: boolean;
  syncStatus: SyncStatus;
  updatedAt: number;
}

export interface ReviewCard {
  card: SrsCard;
  kana: string;
  kanji: string | null;
  meaning: string;
  romaji: string;
}

export interface OverlayLabel {
  left: number;
  top: number;
  width: number;
  height: number;
  original: string;
  translated: string;
}
